go.mod

```diff
@@ -1,6 +1,6 @@
 module github.com/hashicorp/packer-plugin-amazon
 
-go 1.21.0
+go 1.21.13
```

Comment:
Is there a reason to bump this? `.go-version` changes which version of go is used to build the plugin binaries, so by bumping this, the next releases will be built with this version, as for the `go` directive here, I'd be tempted to leave it at `1.21.0` since it's the minimum version you have to use to build the plugin as of today.

Reply:
@lbajolet-hashicorp thanks a lot for the review. 🙌  I removed this part for now as we discussed internally to tackle go bumps in a second PR